### PR TITLE
Add Application Insights smart detection config api

### PR DESCRIPTION
**Pull request by aviled**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5643 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5643/commits/c58d09d6a22fa9287a6cb83c746c9333ffd4f8cf

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+            },
+            "Description": {
+              "type": "string",
+              "readOnly": false,
+              "description": "The rule description"
+            },
+            "HelpUrl": {
+              "type": "string",
+              "readOnly": false,
+              "description": "URL which displays aditional info about the proactive detection rule"
```
Comment (nschonni) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]:
```suggestion
              "description": "URL which displays additional info about the proactive detection rule"
```

**Comment by aviled** [comment]
@tmilos is added to the review. #Closed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json (hunk 1 above)
Comment (aviled) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]: done.

---
In reply to: [270709310](https://github.com/Azure/azure-rest-api-specs/pull/5518#discussion_r270709310) [](ancestors = 270709310)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json [hunk 2]
```diff
@@ -0,0 +1,315 @@
+{
```
Comment (sarangan12) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]: This file is never referenced from the readme file. So, there is no validation running on it. Please add a suitable tag and update this file in the readme file. So, I can review the same.

**Review by sarangan12 — CHANGES_REQUESTED** [review]

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json (hunk 2 above)
Comment (aviled) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]: sorry, can you please explain where should I reference this file exactly? 

---
In reply to: [271035135](https://github.com/Azure/azure-rest-api-specs/pull/5518#discussion_r271035135) [](ancestors = 271035135)

Comment (sarangan12) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]: Refer example here: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/applicationinsights/resource-manager/readme.md#tag-package-2018-06-17-preview

Comment (aviled) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json]: Done.

---
In reply to: [271504122](https://github.com/Azure/azure-rest-api-specs/pull/5518#discussion_r271504122) [](ancestors = 271504122)

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**